// react-text-gradients/lib/cjs/helpers/transformGradient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toGradientStyle = exports.toLinearStyle = exports.toRadialStyle = exports.defaultStyles = exports.directions = void 0;
exports.directions = [
    'to left',
    'to top left',
    'to bottom left',
    'to right',
    'to top right',
    'to bottom right',
    'to top',
    'to bottom',
];
exports.defaultStyles = {
    WebkitBackgroundClip: 'text',
    WebkitTextFillColor: 'transparent',
    WebkitBoxDecorationBreak: 'clone',
};
const genDefaultStyle = (fallbackColor) => fallbackColor
    ? { ...exports.defaultStyles, backgroundColor: fallbackColor }
    : exports.defaultStyles;
const toRadialStyle = (gradient, fallbackColor) => {
    if (!Array.isArray(gradient))
        throw new Error(`Expected gradient to be of type (gradient={[<color values>]), got '${gradient}'`);
    return {
        ...genDefaultStyle(fallbackColor),
        backgroundImage: `radial-gradient(${gradient})`,
    };
};
exports.toRadialStyle = toRadialStyle;
const toLinearStyle = (gradient, fallbackColor) => {
    const [dir, colors] = gradient;
    const direction = exports.directions.includes(dir) ? dir : null;
    if (!direction)
        throw new Error(`Expected gradient to be of type (gradient={["<direction>", [<color values>]]), got '${gradient}'`);
    return {
        ...genDefaultStyle(fallbackColor),
        backgroundImage: `linear-gradient(${direction}, ${colors})`,
    };
};
exports.toLinearStyle = toLinearStyle;
const toGradientStyle = (gradient, gradientType, fallbackColor) => {
    if (gradientType === 'radial')
        return (0, exports.toRadialStyle)(gradient, fallbackColor);
    if (gradientType === 'linear')
        return (0, exports.toLinearStyle)(gradient, fallbackColor);
};
exports.toGradientStyle = toGradientStyle;
